### PR TITLE
MAINT: Update sol2

### LINKED ISSUE #16: BUG,BLD: Fix / cleanup the CI
Currently the runs fail on valid PRs which is confusing.

### PR DESCRIPTION
**Pull request by RuhiRG**
This should close #16 if it works. Also made a small update to the documentation, but there is more to do there.

### CODE REVIEW

**Review by amritagos — APPROVED** [review]
LGTM. Thank you @RuhiRG! @HaoZeke `nix` should be rewritten in the documentation.

File: .github/workflows/build_lib.yml [hunk 1]
```diff
@@ -9,11 +9,24 @@ jobs:
...
+          mkdir bbdir
+          cd bbdir
+          cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_EXPORT_COMPILE_COMMANDS=YES -DCMAKE_INSTALL_PREFIX:PATH=$CONDA_PREFIX ../
+          make -j$(nproc)
+          make install
+
+      - name: Run a test
+        shell: bash -l {0}
+        run: |
+          yodaStruct -c lua_inputs/config.yml
```
Comment (HaoZeke) [inline .github/workflows/build_lib.yml:32]:
```suggestion
```

Shouldn't be duplicated :)

File: .github/workflows/build_pkg.yml [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
-    - run: echo OK
+      - uses: actions/checkout@v3
+
+      - name: Install Conda environment from environment.yml
+        uses: mamba-org/provision-with-micromamba@main
+
+      - name: Build application with cmake
+        shell: bash -l {0}
+        run: |
+          mkdir bbdir
```
Comment (HaoZeke) [inline .github/workflows/build_pkg.yml:15]:
```suggestion
        run: |
          luarocks install luafilesystem
          mkdir bbdir
```

As documented in the `readme`.

**Review by HaoZeke — APPROVED** [review]
Two very minor nits aside this looks great to me. Thanks so much.

**Pull request merged**